backend/backend_mantid_reducer.py

```diff
@@ -70,23 +67,116 @@ def _process_event_data(self, data):
+   
+    def _delete_from_dict(self, transition):
+	new_dict= {}
+	for key, value in self.transition_objects_dict.items():
+	    object_list=[]
+	    for transition_object in value:
+		if transition_object != transition:
+		    object_list.append(transition_object)
+	    new_dict[key] = object_list	
+	self.transition_objects_dict = new_dict
+
+
+    def _delete_transition_object(self, transition):
+	for transitions in transition._transitions:
+	    self._delete_transition_object(transitions)
+	    self._delete_from_dict(transitions)
+	    print "deleting: " + str(transitions)
+	    del transitions
+		
+    def _remove_from_parents_transitions_list(self):
+	parent_transition = self.target_transition._parents[0] #only 1 parent
+	print parent_transition
+	updated_transitions = []
+	for transition in parent_transition()._transitions:
+	    if transition != self.target_transition:
+		updated_transitions.append(transition)
+	parent_transition()._transitions = updated_transitions	
+
+
+    def _delete_transition(self, transition_id):
+	self.find_transition(transition_id)
+	for transitions in self.target_transition._transitions:
+	    self._delete_from_dict(transitions)
+	self._delete_from_dict(self.target_transition)
+	self._remove_from_parents_transitions_list()
+	self.target_transition._transitions = []
+
+    @property
+    def transition_tree(self):
+	self.tree_string = ""
+	self.tree(self._create_workspace_from_events_transition, '')
+	self.tree_string += "\n \n REMEMBER '-' separates \n To add new transition '[parentID]-[transitionType]' \n To delete '[transitionID]-DEL' (This will delete all children) \n To update parameters '[transitionID]-[newParams]'"
+	return self.tree_string
+	
+    @transition_tree.setter
+    def transition_tree(self, new_transition):
+	self._lock.acquire()
+	if new_transition.split('-')[1] == 'DEL':
+	    self._delete_transition(new_transition.split('-')[0])
+	else:
+	    new_transition_object, transition_type = self.add_transition(new_transition)
+	    self.post_transition_settings(new_transition_object, transition_type)	
+	    self.transition_objects_dict[transition_type].append(new_transition_object)
+
+	self._lock.release() #TODO add in automated 'if nothing entered, just set up what is expected' thing
 
     @property
     def bin_parameters(self):
-        return self._rebin_transition._bin_parameters
+	bin_parameters_string = ""
+	for transition in self.transition_objects_dict['MantidRebin']:
+	    bin_parameters_string += "\n" + transition.get_name() +"-" + transition._bin_parameters
+	return bin_parameters_string
 
     @bin_parameters.setter
     def bin_parameters(self, parameters):
         self._lock.acquire()
-        self._rebin_transition.set_bin_parameters(parameters)
+	transition_ID = parameters.split('-')[0] #make sure to strip spaces from this?
+	actual_parameters = parameters.split('-')[1]
+	self.find_transition(transition_ID)
+        self.target_transition.set_bin_parameters(actual_parameters)
```

Comment:
These 4 lines now show up repeatedly. We should keep this in mind for refactoring and later provide a cleaner mechanism to avoid this duplication.